contents/docs/postgres-support/index.mdx

```diff
@@ -92,24 +92,36 @@ Tables with relationships to themselves (i.e., `comment` that can have a parent
 2. Our `createTableSchema` helper can’t deal with recursive types, so you have to first define the object outside `createTableSchema` then pass it as an arg, like so:
 
    ```tsx
-   const foo = {
-     tableName: 'issue',
+   import { TableSchema } from '@rocicorp/zero';
+
+   const comment = {
+     tableName: 'comment',
      columns: {
        id: {type: 'string'},
        title: {type: 'string'},
+       parentID: {type: 'string', optional: true},
      },
      primaryKey: ['id'],
      relationships: {
-       self: {
-         source: 'id',
-         dest: {
-           field: 'id',
-           schema: () => issueSchema,
-         },
+       parent: {
+         sourceField: 'parentID',
+         destField: 'id',
+         // without the any the whole table definition would also be any
+         destSchema: (): any => commentSchema,
+       },
+       children: {
+         sourceField: 'id',
+         destField: 'parentID',
+         destSchema: (): any => commentSchema,
        },
      },
-   } as const;
-   const issueSchema = createTableSchema(foo);
+   } as const satisfies TableSchema;
+   const commentSchema = createTableSchema(comment);
+
+   // get comment by id with comments 2 levels up
+   z.query.comment
+   .where('id', '=', '*id*')
+   .related('parent', (q) => q.related('parent'));
```

Comment:
You can get away with the `any` types by doing it like:

```ts
import {createSchema, Zero} from '@rocicorp/zero';

const commentSchema = {
  tableName: 'comment',
  columns: {
    id: {type: 'string'},
    title: {type: 'string'},
    parentID: {type: 'string', optional: true},
  },
  primaryKey: ['id'],
  relationships: {
    parent: {
      sourceField: 'parentID',
      destField: 'id',
      destSchema: () => commentSchema,
    },
    children: {
      sourceField: 'id',
      destField: 'parentID',
      destSchema: () => commentSchema,
    },
  },
} as const;

const zero = new Zero({
  userID: 'anon',
  schema: createSchema({
    version: 1,
    tables: {
      comment: commentSchema,
    },
  }),
});

const id = 'some-id';
zero.query.comment
  .where('id', '=', id)
  .related('parent', q => q.related('parent'));
```

Reply:
Fixed it in the PR. Thanks!

Although without the TableSchema you lose type safety for the table definition. So in practice you'd probably use it when editing the tables.